ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -30,68 +32,70 @@
+        FloatInput flywheelTargetVelocity = shooterStates.selectByState(FloatInput.zero, // passive
+                FloatInput.zero, // ejecting
+                flywheelLowSpeed, // intaking
+                FloatInput.zero,
+                FloatInput.zero,
+                flywheelHighSpeed,
+                flywheelHighSpeed);
+        
+        PIDTalon flywheelTalon = new PIDTalon(makeLinkedTalons(), "Shooter Flywheel", flywheelTargetVelocity.withRamping(15.0f, FRC.constantPeriodic));
+        flywheelTalon.setup(BooleanInput.alwaysTrue);
+        
+        BooleanInput inhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Intake");
+        BooleanInput exhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Eject");
+        BooleanInput prefireButton = ZukoAzula.controlBinding.addBoolean("Shooter Spinup");
+        BooleanInput fireButton = ZukoAzula.controlBinding.addBoolean("Shooter Fire");
+        BooleanInput cancelShooterButton = ZukoAzula.controlBinding.addBoolean("Shooter Cancel Action");
+        
+        // Behavior
+        shooterStates.setStateWhen("passive", cancelShooterButton.onPress());
+        shooterStates.setStateWhen("ejecting", exhaleButton.onPress());
+        shooterStates.setStateWhen("intaking", inhaleButton.onPress());
+        shooterStates.transitionStateWhen("intaking", "loaded", flywheelTalon.speed.atMost(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Maximum Off Speed", 10.0f)).onPress());
+        shooterStates.setStateWhen("cocking", prefireButton.onPress());
+        shooterStates.setStateWhen("firing", fireButton.onPress()
+                .and(flywheelTalon.speed.atLeast(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Minimum High Speed", 2300.0f))));
+       
+        shooterStates.setStateWhen("passive", FRC.startDisabled);
+        shooterStates.setStateWhen("passive", FRC.startTele);
+        shooterStates.setStateWhen("passive", FRC.startAuto);
+        shooterStates.setStateWhen("passive", FRC.startTest);
+        
+        // turn off cocking after timer expires
+        PauseTimer preloadingTimer = new PauseTimer(ZukoAzula.mainTuning.getFloat("Shooter Cocking Timer", 0.12f));
+        preloadingTimer.triggerAtEnd(shooterStates.getStateTransitionEvent("cocking", "spinup"));   
+        shooterStates.onEnterState("cocking", preloadingTimer);
+        
+        shooterStates.selectByState(FloatInput.zero, // passive
+                ZukoAzula.mainTuning.getFloat("Shooter Eject Speed", 1.0f), // ejecting
+                ZukoAzula.mainTuning.getFloat("Shooter Intake Speed", -1.0f), // intaking
+                FloatInput.zero, // loaded,
+                FloatInput.always(1.0f), // cocking
+                FloatInput.zero, // spinup
+                FloatInput.always(1.0f)).send(FRC.talonSimpleCAN(7, FRC.MOTOR_FORWARD).combine(FRC.talonSimpleCAN(8, FRC.MOTOR_FORWARD))); // firing
     }
```

Comment:
Also, it would be easier to follow if you stick the rollers into a variable first, instead of trying to put them inline. That would let it be clear what they're actually, because right now it's not specifically stated.